apps/api-extractor/src/api/ExtractorMessage.ts

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as tsdoc from '@microsoft/tsdoc';
+import * as path from 'path';
+import { ExtractorMessageId } from './ExtractorMessageId';
+import { Path, Text } from '@microsoft/node-core-library';
+
+/**
+ * Specifies a category of messages for use with {@link ExtractorMessage}.
+ * @public
+ */
+export const enum ExtractorMessageCategory {
+  /**
+   * Messages originating from the TypeScript compiler.
+   *
+   * @remarks
+   * These strings begin with the prefix "TS" and have a numeric error code.
+   * Example: `TS2551`
+   */
+  Compiler = 'Compiler',
+
+  /**
+   * Messages related to parsing of TSDoc comments.
+   *
+   * @remarks
+   * These strings begin with the prefix "tsdoc-".
+   * Example: `tsdoc-link-tag-unescaped-text`
+   */
+  TSDoc = 'TSDoc',
+
+  /**
+   * Messages related to API Extractor's analysis.
+   *
+   * @remarks
+   * These strings begin with the prefix "ae-".
+   * Example: `ae-extra-release-tag`
+   */
+  Extractor =  'Extractor'
+}
+
+/**
+ * Constructor options for `ExtractorMessage`.
+ */
+export interface IExtractorMessageOptions {
+  category: ExtractorMessageCategory;
+  messageId: tsdoc.TSDocMessageId | ExtractorMessageId | string;
+  text: string;
+  sourceFilePath?: string;
+  sourceFileLine?: number;
+  sourceFileColumn?: number;
+}
+
+/**
+ * This object is used to report an error or warning that occurred during API Extractor's analysis.
+ *
+ * @public
+ */
+export class ExtractorMessage {
+  /**
+   * The category of issue.
+   */
+  public readonly category: ExtractorMessageCategory;
+
+  /**
+   * A text string that uniquely identifies the issue type.  This identifier can be used to suppress
+   * or configure the reporting of issues, and also to search for help about an issue.
+   */
+  public readonly messageId: tsdoc.TSDocMessageId | ExtractorMessageId | string;
```

Comment:
Is `string` here because of the compiler messages? Typescript doesn't export an enum of error messages?

Reply:
Right, the TypeScript errors are number codes such as "TS1234".